grammars/php.cson

```diff
@@ -825,6 +850,13 @@
   }
 ]
 'repository':
+  'primitives':
+    'patterns': [
+      {
+        'match': '\\b(array|bool|callable|float|int|iterable|object|string)\\b'
```

Comment:
@Ingramz I'm not sure how unicode is treated in vscode, but in atom I couldn't abuse this with stuff like `stringʕ•ᴥ•ʔ`. However, this _might_ be a problem in vscode or some other editor.

Although it's rather edge case, replacing `\b` with `^...$` is a good idea.

Reply:
One possible example is using emojis like `int🐱`.

Reply:
Thanks! I basically copy-pasted from other parts of the syntax to start with, so I appreciate this advice.

All the tests pass with `^(...)$` but interestingly, this doesn't work in VS Code. No clue why, it just doesn't tokenize properly. But if I remove the capturing group, so it really is just `^...$` (no parentheses) then it works as expected.

Is there any reason to keep those, or removing them as I've done now fine?

Reply:
Without parentheses you have equivalent of: `(^array)|bool|callable|float|int|iterable|object|(string$)` so only array and string are broken then.

Okay I dived into this and found cause https://github.com/microsoft/vscode-textmate/issues/74 tl;dr only `$` is working somewhat correctly and all because `substring(line start, capture end)` is used for match, thus any start anchors are useless there.

Even if `^..$` are better, implementation makes them useless in vscode.

Reply:
Interesting. So would something like `(^array$)|(^bool$)|(^callable$)` work here? Or would it still break because `^` doesn't work properly at all?

@Ingramz it looks like `\\b` might be simpler after all if that's okay with you! 😂

Reply:
Yes, `^` is broken. And yes, in that case `\b` is better. I wasn't aware of that until now and this explains some weird behaviors which I had in the past.

Reply:
Since it is broken in vscode then we have no other choice than to use `\b`. Thank you for trying it out and @KapitanOczywisty for finding the relevant issue. Rest of the pull request looks great now. I am however annoyed by this vscode's deviation from spec.

Reply:
Yeah that's frustrating. I've changed it back to `\b`.